### PR TITLE
New Resource: `azurerm_automation_watcher`

### PR DESCRIPTION
**Pull request by wuxu92**
support automation_watcher depends on pr of automation_rubook_worker 

--- PASS: TestAccWatcher_basic (329.75s)
--- PASS: TestAccWatcher_update (906.01s)
PASS

### CODE REVIEW

File: internal/services/automation/automation_watcher.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"execution_frequency_in_seconds": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ValidateFunc: nil,
+		},
+		"script_name": {
```
Comment (katbyte) [inline internal/services/automation/automation_watcher.go:67]: white space between properties please
```suggestion
		"resource_group_name": commonschema.ResourceGroupName(),
		
		"automation_account_name": {
			Type:         pluginsdk.TypeString,
			Required:     true,
			ForceNew:     true,
			ValidateFunc: validation.StringIsNotEmpty,
		},
		
		"name": {
			Type:         pluginsdk.TypeString,
			Required:     true,
			ForceNew:     true,
			ValidateFunc: validation.StringIsNotEmpty,
		},
		
		"location": commonschema.Location(),
		
		"tags":     commonschema.Tags(),
		
		"etag": {
			Type:         pluginsdk.TypeString,
			Optional:     true,
			ValidateFunc: validation.StringIsNotEmpty,
		},
		
		"execution_frequency_in_seconds": {
			Type:         pluginsdk.TypeInt,
			Required:     true,
			ValidateFunc: nil,
		},
		
		"script_name": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]
we are missing docs?

**Comment by wuxu92** [comment]
document added and other updates

File: internal/services/automation/automation_watcher_test.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("execution_frequency_in_seconds").HasValue("2"),
+			),
+		},
+		data.ImportStep("tags", "etag", "location"),
+	})
+}
+
+func TestAccWatcher_update(t *testing.T) {
```
Comment (katbyte) [inline internal/services/automation/automation_watcher_test.go:216]: these should be up above, basic being the first test. not at the bottom

File: internal/services/automation/automation_watcher_test.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+		return nil, err
+	}
+	resp, err := client.Automation.WatcherClient.Get(ctx, id.ResourceGroup, id.AutomationAccountName, id.Name)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving Type %s: %+v", id, err)
+	}
+	return utils.Bool(resp.WatcherProperties != nil), nil
+}
+
+func (a WatcherResource) template(data acceptance.TestData) string {
```
Comment (katbyte) [inline internal/services/automation/automation_watcher_test.go:32]: tempalte is always at the bottom

File: internal/services/automation/automation_watcher_test.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    subnet_id                     = azurerm_subnet.test.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_linux_virtual_machine" "test" {
+  name                = "acctestVM-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  size                = "Standard_G5"
```
Comment (katbyte) [inline internal/services/automation/automation_watcher_test.go:95]: Do we need to use this size?

```
------- Stdout: -------
=== RUN   TestAccWatcher_basic
=== PAUSE TestAccWatcher_basic
=== CONT  TestAccWatcher_basic
    testcase.go:110: Step 1/2 error: Error running apply: exit status 1
        
        Error: creating Linux Virtual Machine: (Name "acctestVM-220906160028953594" / Resource Group "acctestRG-auto-220906160028953594"): compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="SkuNotAvailable" Message="The requested VM size for resource 'Following SKUs have failed for Capacity Restrictions: Standard_G5' is currently not available in location 'westus2'. Please try another size or deploy to a different location or different zone. See https://aka.ms/azureskunotavailable for details."
        
          with azurerm_linux_virtual_machine.test,
          on terraform_plugin_test.tf line 62, in resource "azurerm_linux_virtual_machine" "test":
          62: resource "azurerm_linux_virtual_machine" "test" {
        
    testing_new.go:84: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: deleting Network Interface: (Name "acctni-220906160028953594" / Resource Group "acctestRG-auto-220906160028953594"): network.InterfacesClient#Delete: Failure sending request: StatusCode=400 -- Original Error: Code="NicReservedForAnotherVm" Message="Nic(s) in request is reserved for another Virtual Machine for 180 seconds. Please provide another nic(s) or retry after 180 seconds. Reserved VM: /subscriptions/*******/resourceGroups/acctestRG-auto-220906160028953594/providers/Microsoft.Compute/virtualMachines/acctestVM-220906160028953594" Details=[]
        
--- FAIL: TestAccWatcher_basic (55.70s)
FAIL
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/automation/automation_watcher_test.go (hunk 4 above)
Comment (wuxu92) [reply internal/services/automation/automation_watcher_test.go]: yes, it's a required property. the value of size should be related to the subscription/account maybe. in my local env, this test passed. and I changed it to the same as other VM acctests value `Standard_F2`

Comment (wuxu92) [reply internal/services/automation/automation_watcher_test.go]: image

now all tests passed in tc

**Review by katbyte — APPROVED** [review]
Thanks @wuxu92 - LGTM 🦺

**Pull request merged**